Bump Search SDK version to 1.0.0-beta.36-SNAPSHOT; Fix api references publishing

DzmitryFomchyn opened the pull request:
<!-- ⚠️ TEMPLATE ⚠️ -->
<!-- Template for GitHub PR descriptions. Use it as a guide on how to describe your work. Feel free to remove any section when you're opening a PR if you think it does not apply for your committed changes. -->

### Description
This PR:
- Bumps Search SDK version to `1.0.0-beta.36-SNAPSHOT`
- Fixes API references uploading
- Uses passed `SearchPlace.id` as a saved favorite id.


### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR (where applicable)
- [x] I have run tests and automatic checks locally
- [ ] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [x] I have made corresponding changes to the documentation (where applicable)
- [x] I have grouped commits logically or I promise to squash my commits before merge

kalbaxa commented:
so you are doing release publishing as a separate job? that's why parts of current ci are commented out?

sorry, can't get motivation from code and no description.

DzmitryFomchyn commented:
> so you are doing release publishing as a separate job? that's why parts of current ci are commented out?
> 
> sorry, can't get motivation from code and no description.

That was a test commit for checking API references publishing in the CI. I've removed test commit and added a description.

kalbaxa reviewed (approved):
lgtm
